pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/AnswerDao.java

```diff
@@ -300,23 +128,190 @@ private void createAnswerPicklistValue(long instanceId, long answerId, PicklistA
+        jdbiCompositeAnswer.deleteChildAnswerItems(answerId);
+        oldChildRows.stream()
+                .flatMap(Collection::stream)
+                .filter(child -> child != null && child.getId() != null && !allNewChildIds.contains(child.getId()))
+                .forEach(orphaned -> deleteAnswer(orphaned.getId()));
+
+        // Recreate child answer references
+        jdbiCompositeAnswer.insertChildAnswerItems(answerId, newChildIdRows);
+    }
+
+    private void updateAnswerPicklistValue(long answerId, PicklistAnswer newAnswer) {
+        var answerSql = getAnswerSql();
+        var picklistAnswerDao = getPicklistAnswerDao();
+        String instanceGuid = answerSql.findInstanceGuidByAnswerId(answerId)
+                .orElseThrow(() -> new DaoException("Could not find activity instance guid for answer id " + answerId));
+        answerSql.deletePicklistSelectedByAnswerId(answerId);
+        picklistAnswerDao.assignOptionsToAnswerId(answerId, newAnswer.getValue(), instanceGuid);
+    }
+
+    //
+    // deletes
+    //
+
+    default void deleteAnswer(long answerId) {
+        deleteAnswers(Set.of(answerId));
+    }
+
+    default void deleteAnswers(Set<Long> answerIds) {
+        var answerSql = getAnswerSql();
+        var jdbiCompositeAnswer = getJdbiCompositeAnswer();
+
+        Set<Long> childAnswerIds = new HashSet<>();
+        Map<Long, QuestionType> typesByAnswerId = answerSql.findQuestionTypesByAnswerIds(answerIds);
+
+        for (var answerId : answerIds) {
+            QuestionType type = typesByAnswerId.get(answerId);
+            if (type == null) {
+                throw new DaoException("Could not find question type for answer with id " + answerId);
+            }
+            if (type == QuestionType.COMPOSITE) {
+                CompositeAnswerSummaryDto parentDto = jdbiCompositeAnswer.findCompositeAnswerSummary(answerId)
+                        .orElseThrow(() -> new DaoException("Could not find parent composite answer with id " + answerId));
+                parentDto.getChildAnswers().stream()
+                        .flatMap(Collection::stream)
+                        .filter(child -> child != null && child.getId() != null)
+                        .forEach(child -> childAnswerIds.add(child.getId()));
+            }
+        }
+
+        // Delete parent answers first so it de-references the child answers
+        DBUtils.checkDelete(answerIds.size(), answerSql.bulkDeleteAnswersById(answerIds));
+
+        // Assuming child answers are not composites, otherwise can recursively delete
+        DBUtils.checkDelete(childAnswerIds.size(), answerSql.bulkDeleteAnswersById(childAnswerIds));
+    }
+
+    default void deleteAllByInstanceIds(Set<Long> instanceIds) {
+        deleteAnswers(getAnswerSql().findAllAnswerIdsByInstanceIds(instanceIds));
+    }
+
+    //
+    // queries
+    //
+
     @UseStringTemplateSqlLocator
-    @SqlQuery("queryAnswerById")
+    @SqlQuery("findAnswerById")
```

Comment:
Note that I am using the same name for the sql query as the method name. Seems more consistent and less silly this way.